cni/pkg/ambient/informers.go

```diff
@@ -120,31 +121,49 @@ func (s *Server) Reconcile(input any) error {
 			s.AddPodToMesh(pod)
 		}
 	case controllers.EventUpdate:
-		// For update, we just need to handle opt outs
-		newPod := event.New.(*corev1.Pod)
-		oldPod := event.Old.(*corev1.Pod)
-		ns := s.namespaces.Get(newPod.Namespace, "")
-		if ns == nil {
-			return fmt.Errorf("failed to find namespace %v", ns)
-		}
-		wasEnabled := oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
-		nowEnabled := PodRedirectionEnabled(ns, newPod)
```

Comment:
@howardjohn Why these two checks not same?

Reply:
Good question, seems they should keep same

Reply:
It's an optimization mostly.

`PodRedirectionEnabled` checks namespace and pod to see if injection isn't allowed, and applies `constants.AmbientRedirection`.

`wasEnabled` can assume that if `constants.AmbientRedirection` is already there, we can skip checking the other things.

Reply:
Thanks for your explain, I think it's helpful to add this comment.